kube-client/src/config/file_config.rs

```diff
@@ -66,26 +66,30 @@ pub struct Preferences {
 #[cfg_attr(test, derive(PartialEq, Eq))]
 pub struct NamedExtension {
     /// Name of extension
+    #[serde(default)]
     pub name: String,
     /// Additional information for extenders so that reads and writes don't clobber unknown fields
     pub extension: serde_json::Value,
 }
 
 /// NamedCluster associates name with cluster.
-#[derive(Clone, Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
 #[cfg_attr(test, derive(PartialEq, Eq))]
 pub struct NamedCluster {
     /// Name of cluster
+    #[serde(default)]
     pub name: String,
     /// Information about how to communicate with a kubernetes cluster
-    pub cluster: Cluster,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub cluster: Option<Cluster>,
 }
 
 /// Cluster stores information to connect Kubernetes cluster.
-#[derive(Clone, Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
 #[cfg_attr(test, derive(PartialEq, Eq))]
 pub struct Cluster {
     /// The address of the kubernetes cluster (https://hostname:port).
+    #[serde(default)]
```

Comment:
I think these new optional fields should align with existing optional fields and use `Option` + skip serializing if none.

Reply:
Yeah that's reasonable. Just to be sure I understood it correctly: all strings with `default` attribute would become `Option<String>`?

Reply:
Yeah. Also, `#[serde(skip_serializing_if = "Option::is_none")]`, just like other existing optional fields.

Reply:
Done. This PR got bigger than I initially expected, but it seems like everything still works. As I'm new to Rust, I may have not implemented it correctly or might have some inefficiencies, so I appreciate if you could review it again. 

But this config: 
```yaml
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: ...
    server: https://kubernetes.docker.internal:6443
  name:
contexts:
- context:
    cluster: 
    user: 
  name: 
current-context:
kind: Config
preferences: {}
users:
- name:
  user:
    client-certificate-data: ...
    client-key-data: ...
```

Now works on both kubectl and kube-rs. I may have gone too far with this so I'm option for feedback If that's something that should be reverted.

Reply:
It's looking pretty clean to me. Thanks a lot. Only left a nit on fmt things which is failing atm.

Reply:
Done, formatting is fixed